experimental/addons/gitops/definitions/gitops.cue

```diff
@@ -0,0 +1,282 @@
+							bucketName: parameter.oss.bucketName
+							provider: parameter.oss.provider
+							if parameter.oss.region != _|_ {
+								region: parameter.oss.region
+							}
+							_secret
+							_sourceCommonArgs
+						}
+					}
+
+			}
+
+			outputs: {
+				if parameter.imageRepository != _|_ {
+					imageRepo: {
+						apiVersion: "image.toolkit.fluxcd.io/v1beta1"
+						kind: "ImageRepository"
+						metadat: {
+							name: context.name
+							namespace: context.namespace
+						}
+						spec: {
+							image: parameter.imageRepository.image
+							interval: parameter.pullInterval
+							if parameter.imageRepository.secretRef != _|_ {
+								secretRef: name: parameter.imageRepository.secretRef
+							}
+						}
+					}
+
+					imagePolicy: {
+						apiVersion: "image.toolkit.fluxcd.io/v1beta1"
+						kind: "ImagePolicy"
+						metadata: {
+							name: context.name
+							namespace: context.namespace
+						}
+						spec: {
+							imageRepositoryRef: name: context.name
+							policy: parameter.imageRepository.policy
+							if parameter.imageRepository.filterTags != _|_ {
+								filterTags: parameter.imageRepository.filterTags
+							}
+						}
+					}
+
+					imageUpdate: {
+						apiVersion: "image.toolkit.fluxcd.io/v1beta1"
+						kind: "ImageUpdateAutomation"
+						metadata: {
+							name: context.name
+							namespace: context.namespace
+						}
+					}
+					spec: {
+						interval: parameter.pullInterval
+						sourceRed: {
+							kind: "GitRepository"
+							name: context.name
+						}
+						git: {
+							checkout:ref:branch: parameter.git.branch
+							commit: {
+								author: {
+									email: "kubevelabot@users.noreply.github.com"
+									name: "kubevelabot"
+								}
+								if parameter.imageRepository.commitMessage != _|_ {
+									messageTemplate: "Update image automatically.\n" + parameter.imageRepository.commitMessage
+								}
+								if parameter.imageRepository.commitMessage == _|_ {
+									messageTemplate: "Update image automatically."
+								}
+							}
+							push: branch: parameter.git.branch
+						}
+						update: {
+							path: parameter.path
+							strategy: "Setters"
+						}
+					}
+				}
+
+			}
+
+			_secret: {
+				if parameter.secretRef != _|_ {
+					secretRef: {
+						name: parameter.secretRef
+					}
+				}
+			}
+
+			_sourceCommonArgs: {
+				interval: parameter.pullInterval
+				if parameter.timeout != _|_ {
+					timeout: parameter.timeout
+				}
+			}
+
+		}
+
+		if parameter.agent == "argocd" {  // we probably need to find a way to install the argocd resources (the one specified on the getting started page)
+			output: {
+				apiVersion: "argoproj.io/v1alpha1"
+				kind: "Application"
+				metadata: {
+					name: context.name
+					namespace: context.namespace
+				}
+				spec: {
+					components: {
+						project: default
+
+						source: {
+							repoURL: parameter.url
+							targetRevision: HEAD
+							path: ""
+						}
```

Comment:
Is the path come from user?

Reply:
It represents the cluster path. 🤔 Thinking about it now, it could or could not be supplied by the user. I  can make it user-supplied for more flexibility?

Reply:
Vela also have the capability of multi-cluster, maybe we can combine them together.